tests/zfs-tests/tests/functional/channel_program/synctask_core/tst.set_props.zcp

```diff
@@ -0,0 +1,70 @@
+--
+-- This file and its contents are supplied under the terms of the
+-- Common Development and Distribution License ("CDDL"), version 1.0.
+-- You may only use this file in accordance with the terms of version
+-- 1.0 of the CDDL.
+--
+-- A full copy of the text of the CDDL should have accompanied this
+-- source.  A copy of the CDDL is also available via the Internet at
+-- http://www.illumos.org/license/CDDL.
+--
+
+--
+-- Copyright (c) 2017 by Delphix. All rights reserved.
+-- Copyright 2020 Joyent, Inc.
+--
+
+arg = ...
+fs = arg["argv"][1]
+
+pos_props = {}
+neg_props = {}
+
+pos_props["a:prop"] = {"hello"}
+neg_props["notaproperty"] = {"hello"}
```

Comment:
Similar to my previous comment, I think it would be good to add a test that exercises more invalid property names and values - maybe some where they're greater than `ZAP_MAXVALUELEN`?

Reply:
Added